inference-engine/src/mkldnn_plugin/nodes/mkldnn_conv_node.cpp

```diff
@@ -747,4 +760,59 @@ InferenceEngine::Precision MKLDNNConvolutionNode::getRuntimePrecision() const {
     return MKLDNNExtensionUtils::getMaxPrecision(inputPrecisions);
 }
 
+bool MKLDNNConvolutionNode::isNspcAvailable() const {
+    using impl::cpu::x64::mayiuse;
+    // A bunch of heuristics are designed to cut off not optimal nspc convolution applications
+
+    auto inpDims = getParentEdgeAt(0)->getDims().ToSizeVector();
+    auto outDims = getChildEdgeAt(0)->getDims().ToSizeVector();
+    auto ndims = inpDims.size();
+
+    if (isDepthWise()) {
+        // 1d equivalent cases are painfully slow
+        if (1 == inpDims[inpDims.size() - 2]) {
```

Comment:
No. But this seems like a rather rare case, since in most cases we have `[N, C, 1, W]` to simulate a 1d `[N, C, W]` DW convolution. However, considering the existing DW kernel implementation, it will be also slower than blocked for `[N, C, D, 1, W]` cases.